system/lib/libc/musl/src/thread/__timedwait.c

```diff
@@ -73,6 +73,7 @@ int __timedwait_cp(volatile int *addr, int val,
 		double sleepUntilTime = emscripten_get_now() + msecsToSleep;
 		do {
 			if (pthread_self()->cancel) {
+				__pthread_testcancel();
```

Comment:
so the `return` below is never reached, right? It might be helpful to clarify that.

Reply:
Apparently there are at least some tests that depend on that behaviour yes.

I tried completely removing the `return ECANCELED;` and these test fail:

```
posixtest.test_pthread_cond_wait_2_3
posixtest.test_pthread_cond_timedwait_2_6
```

Reply:
I found that reason for this.  `canceldisable` can be in a third state called `PTHREAD_CANCEL_MASKED` which is used in pthread_cond_timedwait.c.  The expectation here seems to be that `__timedwait_cp` will detect cancellation and return ECANCELLED, as we are doing here.  Which is why this is needed.